### PR TITLE
Added `GitCloneFlags` support to the Golang version of Bootstrap

### PR DESCRIPTION
**Pull request by keithpitt**
`git-clone-flags` and `git-clean-flags` can now also be configured in the `.conf` file

### CODE REVIEW

File: agent/bootstrap.go [hunk 1]
```diff
@@ -864,7 +882,15 @@ func (b *Bootstrap) Start() error {
...
 			b.runCommand("git", "remote", "set-url", "origin", b.Repository)
 		} else {
-			b.runCommand("git", "clone", "-v", "--", b.Repository, ".")
+			// `git clone` can't accept options within quote like
+			// `git clone "-v --depth 1", so we need to split them
+			// by spaces and pass them individually.
+			gitCloneFlags := strings.Split(b.GitCloneFlags, " ")
+			gitCloneArguments := []string{"clone"}
+			gitCloneArguments = append(gitCloneArguments, gitCloneFlags...)
+			gitCloneArguments = append(gitCloneArguments, "--", b.Repository, ".")
```
Comment (sj26) [inline agent/bootstrap.go]: What about args/flags with spaces? e.g. I want to specify config during cloning:

```
GIT_CLONE_FLAGS="-v --config 'description = \'buildkite repo\''"
```

So we have a shell arg parser or something?

Comment (keithpitt) [inline agent/bootstrap.go]: Gah, didn't think of that...

Comment (sj26) [inline agent/bootstrap.go]: Here's a handy port of python's shlex: https://github.com/flynn-archive/go-shlex

File: agent/bootstrap.go [hunk 2]
```diff
@@ -864,7 +883,19 @@ func (b *Bootstrap) Start() error {
...
+			// `git clone` can't accept options within quote like
+			// `git clone "-v --depth 1", so we need to split them
+			// and pass them individually.
+			gitCloneFlags, err := shlex.Split(b.GitCloneFlags)
+			if err != nil {
+				exitf("There was an error trying to split `%s` into arguments (%s)", b.GitCloneFlags, err)
+			}
+
+			gitCloneArguments := []string{"clone"}
+			gitCloneArguments = append(gitCloneArguments, gitCloneFlags...)
```
Comment (sj26) [inline agent/bootstrap.go]: Should we make gitCleanFlags do the same thing while we're here?

Comment (sj26) [inline agent/bootstrap.go]: i.e. I _think_ this will currently break:

```
BUILDKITE_GIT_CLEAN_FLAGS="-v -f -q"
```

Comment (keithpitt) [inline agent/bootstrap.go]: Yeah, good call. Fixed!

**Comment by sj26** [comment]
👍

**Pull request merged**